core/controllers/learner_dashboard_test.py

```diff
@@ -238,6 +246,31 @@ def test_get_learner_dashboard_ids(self):
             learner_dashboard_activity_ids['collection_playlist_ids'],
             [self.COL_ID_3])
 
+    def test_get_with_threads_update_thread_summaries(self):
+        self.login(self.OWNER_EMAIL)
+
+        response = self.get_json(feconf.LEARNER_DASHBOARD_DATA_URL)
+        thread_summaries = response['thread_summaries']
+        self.assertEqual(thread_summaries, [])
+
+        self.save_new_default_exploration(
+            self.EXP_ID_1, self.owner_id, title=self.EXP_TITLE_1)
+        feedback_services.create_thread(
+            'exploration', self.EXP_ID_1, self.owner_id, 'a subject',
+            'some text')
+
+        response = self.get_json(feconf.LEARNER_DASHBOARD_DATA_URL)
+        thread_summaries = response['thread_summaries']
+        self.assertEqual(len(thread_summaries), 1)
+        self.assertEqual(thread_summaries[0]['total_message_count'], 1)
+        self.assertEqual(
+            thread_summaries[0]['exploration_title'], self.EXP_TITLE_1)
+        self.assertEqual(thread_summaries[0]['exploration_id'], self.EXP_ID_1)
+        self.assertEqual(thread_summaries[0]['last_message_text'], 'some text')
+        self.assertEqual(
+            thread_summaries[0]['original_author_id'], self.owner_id)
+        self.logout()
```

Comment:
Please also add a check for the thread subject.

Reply:
Done